library/ssl_tls.c

```diff
@@ -666,6 +666,66 @@ static int resize_buffer( unsigned char **buffer, size_t len_new, size_t *len_ol
 
     return 0;
 }
+
+static void handle_buffer_resizing( mbedtls_ssl_context *ssl, int downsizing,
+                                    uint32_t in_buf_new_len,
+                                    uint32_t out_buf_new_len )
+{
+    int modified = 0;
+    size_t written_in = 0, len_offset_in = 0;
+    size_t written_out = 0, iv_offset_out = 0, len_offset_out = 0;
+    if( ssl->in_buf != NULL )
+    {
+        written_in = ssl->in_msg - ssl->in_buf;
+        len_offset_in = ssl->in_len - ssl->in_buf;
+        if( ( downsizing && ssl->in_buf_len > in_buf_new_len && ssl->in_left < in_buf_new_len ) ||
+            ( !downsizing && ssl->in_buf_len < in_buf_new_len ) )
+        {
+            if( resize_buffer( &ssl->in_buf, in_buf_new_len, &ssl->in_buf_len ) != 0 )
```

Comment:
Nothing wrong will happen if we're out of memory, apart from the fact that the reallocation won't happen. The buffers might stay as they were, and everything will work fine. Are you sure you want to stop the flow in such case?

Reply:
I am not sure at all :) , agree with you it is better to continue.
Let's add description to the feature with this behavior explanation , user should know that it can happen.
And comment in the code that allocation is not checked as apart of feature behavior

Reply:
Added.